pinot-segment-local/src/test/java/org/apache/pinot/segment/local/indexsegment/mutable/MutableSegmentImplUpsertComparisonColTest.java

```diff
@@ -131,43 +146,55 @@ public void testUpsertOutOfOrderRecordColumn()
   public void testUpsertIngestion(UpsertConfig upsertConfig)
       throws Exception {
     setup(upsertConfig);
-    ImmutableRoaringBitmap bitmap = _mutableSegmentImpl.getValidDocIds().getMutableRoaringBitmap();
-    // note offset column is used for determining sequence but not time column
-    Assert.assertEquals(_mutableSegmentImpl.getNumDocsIndexed(), 4);
-    Assert.assertFalse(bitmap.contains(0));
-    Assert.assertTrue(bitmap.contains(1));
-    Assert.assertTrue(bitmap.contains(2));
-    Assert.assertFalse(bitmap.contains(3));
+    try {
+      ImmutableRoaringBitmap bitmap = _mutableSegmentImpl.getValidDocIds().getMutableRoaringBitmap();
+      // note offset column is used for determining sequence but not time column
+      Assert.assertEquals(_mutableSegmentImpl.getNumDocsIndexed(), 4);
+      Assert.assertFalse(bitmap.contains(0));
+      Assert.assertTrue(bitmap.contains(1));
+      Assert.assertTrue(bitmap.contains(2));
+      Assert.assertFalse(bitmap.contains(3));
+    } finally {
+      tearDown();
```

Comment:
Couldn't we run that by default using an `@AfterMethod`?

Reply:
Same in other tests. If it is technically possible to close release resources on an `@AfterMethod` tests would be easier to read and more difficult to miss to release resources

Reply:
It 'is' possible but would require rewriting whole test . That is because `public void setup(UpsertConfig upsertConfig)` is called multiple times per test method, and each time allocates new resources.